### PR TITLE
[Benchmark] fix update commits

### PR DESCRIPTION
**Pull request by Dania-Abuhijleh**

### CODE REVIEW

File: benchmark-service/benchmark/benchmark.py [hunk 1]
```diff
@@ -269,13 +277,11 @@ async def github_polling_loop(app):
...
 
 async def on_startup(app):
-    with open(os.environ.get('HAIL_CI_OAUTH_TOKEN', 'oauth-token/oauth-token'), 'r') as f:
-        oauth_token = f.read().strip()
     app['gs_reader'] = ReadGoogleStorage(service_account_key_file='/benchmark-gsa-key/key.json')
     app['github_client'] = gidgethub.aiohttp.GitHubAPI(aiohttp.ClientSession(),
                                                        'hail-is/hail',
                                                        oauth_token=oauth_token)
-    app['batch_client'] = await bc.BatchClient(billing_project='benchmark')
+    app['batch_client'] = await bc.BatchClient(billing_project='test')
```
Comment (jigold) [inline benchmark-service/benchmark/benchmark.py:284]: This needs to be benchmark

File: build.yaml [hunk 2]
```diff
@@ -2040,11 +2040,11 @@ steps:
    dependsOn:
     - default_ns
     - benchmark_image
-    - deploy_router
-    - create_certs
-    - deploy_auth
-    - create_accounts
-    - deploy_batch
+    #- deploy_router
```
Comment (jigold) [inline build.yaml:2043]: You have to uncomment these.

File: benchmark-service/benchmark/benchmark.py [hunk 3]
```diff
@@ -232,17 +235,22 @@ async def update_commits(app):
     for gh_commit in gh_data:
         sha = gh_commit.get('sha')
         log.info(f'for commit {sha}')
-        file_path = f'gs://{HAIL_BENCHMARK_BUCKET_NAME}/benchmark-test/{sha}.json'
+        file_path = f'{BENCHMARK_RESULTS_PATH}/{sha}.json'
         has_results_file = gs_reader.file_exists(file_path)
         batches = [b async for b in batch_client.list_batches(q=f'sha={sha} running')]
```
Comment (jigold) [inline benchmark-service/benchmark/benchmark.py:240]: Can we move this line to the `elif batches` below. We shouldn't check the running batches until we need to.

File: benchmark-service/benchmark/benchmark.py [hunk 4]
```diff
@@ -232,17 +235,22 @@ async def update_commits(app):
...
 
-        if not batches and not has_results_file:
+        if has_results_file:
+            log.info(f'commit {sha} has a results file')
+            batches = [b async for b in batch_client.list_batches(q=f'sha={sha} complete')]
+            batch = batches[-1]
+            assert batch, batches
+        elif batches:
+            batch = batches[-1]
```
Comment (jigold) [inline benchmark-service/benchmark/benchmark.py:248]: Move check for running batches here.

File: benchmark-service/benchmark/benchmark.py [hunk 5]
```diff
@@ -36,6 +36,9 @@
 
 benchmark_data = None
 
+with open(os.environ.get('HAIL_CI_OAUTH_TOKEN', 'oauth-token/oauth-token'), 'r') as f:
```
Comment (jigold) [inline benchmark-service/benchmark/benchmark.py:39]: Why do you need this?

**Review by jigold — CHANGES_REQUESTED** [review]

File: benchmark-service/benchmark/benchmark.py (hunk 1 above)
Comment (Dania-Abuhijleh) [reply benchmark-service/benchmark/benchmark.py:284]: I don't have to wait for #9668 to go in?

File: benchmark-service/benchmark/benchmark.py (hunk 5 above)
Comment (Dania-Abuhijleh) [reply benchmark-service/benchmark/benchmark.py:39]: for the github client

File: benchmark-service/benchmark/benchmark.py [hunk 6]
```diff
@@ -232,17 +235,22 @@ async def update_commits(app):
...
         has_results_file = gs_reader.file_exists(file_path)
-        batches = [b async for b in batch_client.list_batches(q=f'sha={sha} running')]
 
-        if not batches and not has_results_file:
+        if has_results_file:
+            log.info(f'commit {sha} has a results file')
+            batches = [b async for b in batch_client.list_batches(q=f'sha={sha} complete')]
+            batch = batches[-1]
+            assert batch, batches
+        elif [b async for b in batch_client.list_batches(q=f'sha={sha} running')]:
```
Comment (jigold) [inline benchmark-service/benchmark/benchmark.py:246]: Actually, this is a better way. Can you double check with dev deploy? Not sure I got the batch result structure correct to extract the state.

```
batches = [b async for b in batch_client.list_batches(q=f'sha={sha}')]
complete_batches = [b for b in batches if b['state'] == 'complete']
running_batches = [b for b in batches if b['state'] == 'running']

if has_results_file:
    batch = complete_batches[-1]
elif running_batches:
    batch = running_batches[-1]
else:
    ...
``` 


if has_results_file:

else:
     batches = [b async for b in batch_client.list_batches(q=f'sha={sha} running')]

File: benchmark-service/benchmark/benchmark.py (hunk 5 above)
Comment (jigold) [reply benchmark-service/benchmark/benchmark.py:39]: I understand that, but why did you not have it before and it seemed to be working. Oh. I see. Why did you move it?

**Review by jigold — DISMISSED** [review]

File: benchmark-service/benchmark/benchmark.py (hunk 5 above)
Comment (Dania-Abuhijleh) [reply benchmark-service/benchmark/benchmark.py:39]: I was getting some weird session closed errors, then after I moved it they went away

Comment (jigold) [reply benchmark-service/benchmark/benchmark.py:39]: ok. that's what CI does.

**Review by jigold — APPROVED** [review]

**Pull request merged**